client/src/components/Sidebar/ChatContent.js

```diff
@@ -18,24 +18,63 @@ const useStyles = makeStyles((theme) => ({
     color: "#9CADC8",
     letterSpacing: -0.17,
   },
+  unreadMessageBox: {
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+    height: "100%",
+    margin: "auto 0",
+  },
+  unreadBadge: {
+    height: "20px",
+    width: "20px",
+    borderRadius: "50%",
+    top: "10px",
+    right: "20px",
+  },
 }));
 
+const countUnReadMessage = (messages, otherUserId) =>
+  messages.reduce(
+    (acc, message) =>
+      acc +
+      (message.readStatus === false && message.senderId === otherUserId
+        ? 1
+        : 0),
+    0
+  );
+
 const ChatContent = (props) => {
   const classes = useStyles();
 
-  const { conversation } = props;
+  const { conversation, setUnReadMessage, unReadMessage } = props;
   const { latestMessageText, otherUser } = conversation;
 
+  useEffect(() => {
+    if (!setUnReadMessage || !conversation) return;
+    setUnReadMessage(
+      countUnReadMessage(conversation.messages, conversation.otherUser.id)
+    );
+  }, [conversation, setUnReadMessage]);
+
   return (
     <Box className={classes.root}>
       <Box>
         <Typography className={classes.username}>
           {otherUser.username}
         </Typography>
         <Typography className={classes.previewText}>
-          {latestMessageText}
+          <i>{conversation.isTyping ? "...Typing" : latestMessageText}</i>
```

Comment:
The preview text should not be italicized - if you look at the spec, it shows the preview text bolded if there are unread messages, otherwise its regular weight.

Reply:
noted !